### PR TITLE
Remove Unity; Add Rex

### PR DESCRIPTION
**Pull request by cwfitzgerald**
This is it. This is the end of the unity. But the beginning of rex.

The king is dead; long live the king!

This is just the bare minimum "rex compiles on CI" changes. Actual C api, bindgen, etc will all come later.

### CODE REVIEW

File: bve-rex-sys/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (DethRaid) [inline bve-rex-sys/src/lib.rs]: Do you really need this empty file?

File: bve-rex/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
Comment (DethRaid) [inline bve-rex/src/lib.rs]: Ew another empty file

File: bve-rex-sys/src/lib.rs (hunk 1 above)
Comment (cwfitzgerald) [reply bve-rex-sys/src/lib.rs]: Yes

File: bve-rex/src/lib.rs (hunk 2 above)
Comment (cwfitzgerald) [reply bve-rex/src/lib.rs]: Yes

File: bve-rex-sys/src/lib.rs (hunk 1 above)
Comment (DethRaid) [reply bve-rex-sys/src/lib.rs]: ugh

**Pull request merged**